### PR TITLE
feat: store block and hash info

### PR DESCRIPTION
**Pull request by krobi64**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

References #707

Creates a basic store to save blockchain info and to find the block with the greatest height. This will be called in BlockchainWriter onBlockDownloaded so that we can create a service to verify writes to the blockchain. Once we more fully develop the verification process, more functions will be added to the DAO.

### CODE REVIEW

File: tests/helpers/database.ts [hunk 1]
```diff
@@ -35,6 +36,22 @@ export const dbHelper = () => {
...
+    let result: any
+
+    try {
+      const db = mongoClient.db()
+      result = await pipeP(...fns)(db)
+    } catch (err) {
+      console.log(err)
+    }
+    await delay(1300)
+    return { result, mongoClient }
```
Comment (krobi64) [inline tests/helpers/database.ts]: FYI: mongoClient cannot be closed until after the test runs, so I'm passing this back.

File: src/BlockchainWriter/BlockInfoDAO.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  readonly start = async () => {
+    await this.blockchainInfo.createIndex({ height: 1, hash: 1 }, { unique: true })
+  }
+
+  readonly insertBlockInfo = (lightBlock: LightBlock) =>
+    this.blockchainInfo.insertOne(lightBlock)
+
+  readonly getMaxHeightBlock = async (): Promise<LightBlock> => {
+    const maxBlock = this.blockchainInfo.find().sort({height: -1}).limit(1)
```
Comment (kennylavender) [inline src/BlockchainWriter/BlockInfoDAO.ts]: For slightly better perf:

```suggestion
    const maxBlock = this.blockchainInfo.find({}, { height: 1, hash: 1, _id: 0 }).sort({height: -1}).limit(1)
```

For details see the comment of https://stackoverflow.com/a/32077240 and their link to:
https://docs.mongodb.com/v3.0/core/query-optimization/#covered-query

File: src/BlockchainWriter/BlockInfoDAO.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/* tslint:disable:no-relative-imports */
+import { describe } from 'riteway'
+import { runtimeId, setUpDb } from '../../tests/helpers/utils'
+import { BlockInfoDAO } from './BlockInfoDAO'
+
+const prefix = runtimeId()
```
Comment (lautarodragan) [inline src/BlockchainWriter/BlockInfoDAO.test.ts]: @warrenv is this a unit test or an integration test?

Comment (krobi64) [reply src/BlockchainWriter/BlockInfoDAO.test.ts]: So here's my plan:
1. Keep this as a unit test until it's wired up in the router.
2. Once that's done, move this to an integration test
3. Finally, once #707 is done, move it into a functional test.

Comment (lautarodragan) [reply src/BlockchainWriter/BlockInfoDAO.test.ts]: Got it.

File: src/BlockchainWriter/BlockInfoDAO.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  private readonly blockchainInfo: Collection
+
+  constructor(
+    @inject('BlockchainInfoCollection') blockchainInfo: Collection,
+  ) {
+    this.blockchainInfo = blockchainInfo
+  }
+
+  readonly start = async () => {
+    await this.blockchainInfo.createIndex({ height: 1, hash: 1 }, { unique: true })
```
Comment (lautarodragan) [inline src/BlockchainWriter/BlockInfoDAO.ts]: Just in case, won't we need to support finding by hash only too?

Comment (krobi64) [reply src/BlockchainWriter/BlockInfoDAO.ts]: yes. Will be adding that in a separate PR

**Pull request merged**